nav2_util/include/nav2_util/simple_action_server.hpp

```diff
@@ -74,6 +84,41 @@ class SimpleActionServer
       handle_accepted);
   }
 
+  void activate()
+  {
+    std::lock_guard<std::mutex> lock(activation_mutex_);
+    server_active_ = true;
+  }
+
+  void deactivate()
+  {
+    std::lock_guard<std::mutex> lock_server_state(activation_mutex_);
```

Comment:
I feel like two mutexes in this one class is too fine-grained locking. Let's just use the one mutex to guard all changes to state of this class. Having to take two mutexes is a deadlock waiting to happen, IMHO.

Reply:
Good idea, as mentioned above. The only consequence of this is that handle_goal and handle_accepted would not run simultaneously. Which I think is fine for a simple action server.